subsys/testsuite/ztest/src/ztest.c

```diff
@@ -339,7 +339,7 @@ void main(void)
 		if (test_status == 0) {
 			PRINT("Reset board #%u to test again\n",
 				state.boots);
-			k_sleep(K_MSEC(10));
+			k_sleep(10);
```

Comment:
This blows my mind really, so we have the chance to unify APIs that takes a timeout but instead we decide some APIs such k_sleep should operate in a different unit? I hope this is not to make k_usleep still useful since we could replace that with a k_sleep that can sleep in any unit the user wants instead of creating API variants for each unit.

Reply:
The decision on k_sleep() was to keep it a simple API.  And for a while there seemed to be consensus on that.  But it's not hard to swap it if that's what it takes to get this in.

Reply:
I certainly vote for `k_sleep()` to take a timeout, I just don't see what the real benefit of keeping an old API serves when looking into the future, and especially having an LTS.